test-suite/js/src/test/scala/org/scalajs/testsuite/library/FinalizationRegistryTest.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.library
+
+import org.junit.Assert._
+import org.junit.Assume.assumeTrue
+import org.junit.{BeforeClass, Test}
+import org.scalajs.testsuite.utils.Platform.executingInNodeJS
+
+import scala.scalajs.js
+import scala.scalajs.js.FinalizationRegistry
+
+object FinalizationRegistryTest {
+  @BeforeClass def assumeNodeJs(): Unit = {
+    assumeTrue("Assume Node.js", executingInNodeJS)
+  }
+}
+
+class FinalizationRegistryTest {
+  @Test def testConstructor(): Unit = {
+    assertNotNull(new js.FinalizationRegistry[Any, String, Any]((heldValue: String) => {}))
+  }
+
+  @Test def testMethods(): Unit = {
+    val registry = new FinalizationRegistry[js.Date, String, Any]((heldValue: String) => {})
+
+    val obj = new js.Date()
+    val ungregisterToken = new js.Object()
+
+    registry.register(obj, "foo")
+    registry.register(obj, "bar", obj)
+    registry.register(obj, "bar", ungregisterToken)
```

Comment:
It seems weird to register the same `obj` several times. Consider using three different `obj1`, `obj2` and `obj3`.